libs/libuv/stone.yml

```diff
@@ -0,0 +1,26 @@
+#
+# SPDX-FileCopyrightText: © 2020-2023 Serpent OS Developers
+#
+# SPDX-License-Identifier: Zlib
+#
+name        : libuv
+version     : '1.44.2'
+release     : 1
+homepage    : https://libuv.org/
+upstreams   :
+    - https://dist.libuv.org/dist/v1.44.2/libuv-v1.44.2.tar.gz : ccfcdc968c55673c6526d8270a9c8655a806ea92468afcbcabc2b16040f03cb4
+summary     : Cross-platform asynchronous I/O library
+description : |
+    libuv is a multi-platform support library with a focus on asynchronous I/O.
+license     :
+    - MIT
+    - ISC
+    - BSD-2-Clause
+setup       : |
+    %cmake
+build       : |
+    %cmake_build
```

Comment:
Looks like there are tests for this package

Reply:
I really can't get them to run :/ When I setup with `%cmake -DBUILD_TESTING=ON` and then run `serpent_builddir/uv_run_tests` as [recommended here](https://github.com/libuv/libuv#build-instructions), I get 
```
The libuv test suite cannot be run as root.
```

When I tried the autogen way, it wouldn't even build:
```
+ libtoolize --copy --force
libtoolize: putting macros in AC_CONFIG_AUX_DIR, '/mason/build/build/x86_64/libuv-v1.44.2/build-aux'.
libtoolize: copying file '/mason/build/build/x86_64/libuv-v1.44.2/build-aux/ltmain.sh'
...
configure.ac:44: error: required file './ltmain.sh' not found
```

Reply:
What did the autotools version of the build script look like? I think slibtool is supposed to provide that file but it should be pulled in automatically by `autoconf` which is an automatically added if you use `%configure`, `%reconfigure`, or `%autogen` macros.

Reply:
autotools version looks like this
```yaml
name        : libuv
version     : '1.44.2'
release     : 1
homepage    : https://libuv.org/
upstreams   :
    - https://dist.libuv.org/dist/v1.44.2/libuv-v1.44.2.tar.gz : ccfcdc968c55673c6526d8270a9c8655a806ea92468afcbcabc2b16040f03cb4
summary     : Cross-platform asynchronous I/O library
description : |
    libuv is a multi-platform support library with a focus on asynchronous I/O.
license     :
    - BSD-2-Clause
    - ISC
    - MIT
setup       : |
    %autogen
build       : |
    %make
install     : |
    %make_install
    rm -f %(installroot)%(libdir)/pkgconfig/libuv-static.pc
    rm -f %(installroot)%(libdir)/libuv_a.a
tuning      :
    - lto
    - optimize: speed
```

Reply:
I was able to get the tests to run with the autotools version. Note that I changed the source url here to one that was pre-processed (IE it has a `configure`, not just a `configure.ac`). 

```
name        : libuv
version     : '1.44.2'
release     : 1
homepage    : https://libuv.org/
upstreams   :
    - https://dist.libuv.org/dist/v1.44.2/libuv-v1.44.2-dist.tar.gz : 8ff28f6ac0d6d2a31d2eeca36aff3d7806706c7d3f5971f5ee013ddb0bdd2e9e
summary     : Cross-platform asynchronous I/O library
description : |
    libuv is a multi-platform support library with a focus on asynchronous I/O.
license     :
    - BSD-2-Clause
    - ISC
    - MIT
networking  : yes # Needed for tests
checkdeps   :
    - binary(setpriv)
setup       : |
    %configure --disable-static
build       : |
    %make
install     : |
    %make_install
    rm -vf %(installroot)%(libdir)/pkgconfig/libuv-static.pc
    rm -vf %(installroot)%(libdir)/libuv_a.a
check       : |
    # This will error out if you try to run it as root
    # TODO Tests 46 47 48 75 110 120 208 251 252 fail, likely due to missing patches in glibc for statelessness
    setpriv --reuid=65532 --regid=65532 --clear-groups %make check || true
tuning      :
    - lto
    - optimize: speed
```
Depends on #38

Reply:
Also I didn't check if those `rm` commands could be removed.